src/Tooltip2.tsx

```diff
@@ -0,0 +1,257 @@
+        }),
+      ...(wrap &&
+        (direction === 'w' || direction === 'e') && {
+          right: '100%',
+        }),
+    },
+    '&::before': {
+      display: 'inline-block',
+      textDecoration: 'none',
+      // Left align tooltips with align prop
+      ...(align === 'left' && {
+        left: '10px',
+      }),
+      // Right align tooltips with align prop
+      ...(align === 'right' && {
+        right: '15px',
+      }),
+      //   conditionally render styles depending on direction
+      ...((direction === 'n' || direction === 'ne' || direction === 'nw') && {
+        borderTopColor: 'neutral.emphasisPlus',
+        top: '-7px',
+        bottom: 'auto',
+        right: '50%',
+        marginRight: '-6px',
+      }),
+      ...((direction === 's' || direction === 'se' || direction === 'sw') && {
+        borderBottomColor: 'neutral.emphasisPlus',
+        top: 'auto',
+        bottom: '-7px',
+        right: '50%',
+        marginRight: '-6px',
+      }),
+      ...(direction === 'e' && {
+        borderRightColor: 'neutral.emphasisPlus',
+        top: '50%',
+        right: '-7px',
+        bottom: '50%',
+        marginTop: '-6px',
+      }),
+      ...(direction === 'w' && {
+        borderLeftColor: 'neutral.emphasisPlus',
+        top: '50%',
+        bottom: '50%',
+        left: '-7px',
+        marginTop: '-6px',
+      }),
+    },
+  },
+})
+
+const Tooltip2: React.FC<React.PropsWithChildren<Tooltip2Props>> = ({
+  direction = 'n',
+  text,
+  noDelay,
+  align,
+  wrap,
+  sx = {},
+  children,
+  type = 'label',
+  ...props
+}) => {
+  const id = useId()
+  const tooltipRef = useRef<HTMLDivElement>(null)
+  const child = Children.only(children) // make sure there is only one child
+
+  // we need this check for every render
+  useEffect(() => {
+    if (tooltipRef.current) {
+      const childNode = tooltipRef.current.children[0] // For now, I assume it has one node but that is not true
+      if (!isFocusable(childNode as HTMLElement)) {
+        throw new Error('Tooltip2: The child element must be focusable')
+      }
+    }
+  })
+  const triggerProps = {
+    // if it is a type description, we use tooltip to describe the trigger
+    'aria-describedby': type === 'description' ? id : undefined,
+    // If it is a type description, we should keep the aria label if it exists, otherwise we remove it because we will use aria-labelledby
+    'aria-label': type === 'description' ? (children as React.ReactElement).props['aria-label'] : undefined,
+    //   If it is a label type, we use tooltip to label the trigger
+    'aria-labelledby': type === 'label' ? id : undefined,
+  }
+  // Only need tooltip role if the tooltip is a description for supplementary information
+  const role = type === 'description' ? 'tooltip' : undefined
+  // aria-hidden true only if the tooltip is a label type
+  const ariaHidden = type === 'label' ? true : undefined
+
+  return (
+    <>
+      <style>
```

Comment:
I need feedback on this 👀 I couldn't find a way to declare animations without using `keyframes` from `styled-components` so I ended up doing this.

Reply:
I was trying to remove the styled components completely considering that we are looking into moving away from it but without having an alternative, it doesn't make much sense. I found a couple of possible ways when I searched on google but it doesn't worth the effort, I don't think. I'll revert this back to use `styled-components` for now.